app/scripts/controllers/transactions/index.js

```diff
@@ -506,7 +506,9 @@ export default class TransactionController extends EventEmitter {
   async _getDefaultGasFees(txMeta, eip1559Compatibility) {
     if (
       (!eip1559Compatibility && txMeta.txParams.gasPrice) ||
-      (txMeta.txParams.maxFeePerGas && txMeta.txParams.maxPriorityFeePerGas)
```

Comment:
This fixes a bug on develop. Without this `eip1559Compatibility` compatibility check, default gas fees could be returned when using a non-EIP1559-compatible wallet